src/hotspot/cpu/riscv/riscv_v.ad

```diff
@@ -2951,7 +3854,440 @@ instruct rearrange(vReg dst, vReg src, vReg shuffle) %{
+  format %{ "insertD_index_lt32 $dst, $src, $val, $idx" %}
+  ins_encode %{
+    __ vsetvli_helper(T_DOUBLE, Matcher::vector_length(this));
+    __ vid_v(as_VectorRegister($v0$$reg));
+    __ vadd_vi(as_VectorRegister($v0$$reg), as_VectorRegister($v0$$reg), -16);
+    __ vmseq_vi(as_VectorRegister($v0$$reg), as_VectorRegister($v0$$reg), (int)($idx$$constant) - 16);
+    __ vfmerge_vfm(as_VectorRegister($dst$$reg), as_VectorRegister($src$$reg), $val$$FloatRegister);
+  %}
+  ins_pipe(pipe_slow);
+%}
+
+instruct insertD_index(vReg dst, vReg src, fRegD val, iRegIorL2I idx, vReg tmp1, vRegMask_V0 v0) %{
+  predicate(n->in(2)->get_int() >= 32 &&
+            (Matcher::vector_element_basic_type(n) == T_DOUBLE));
+  match(Set dst (VectorInsert (Binary src val) idx));
+  effect(TEMP tmp1, TEMP v0);
+  format %{ "insertD_index $dst, $src, $val, $idx\t# KILL $tmp1" %}
+  ins_encode %{
+    __ vsetvli_helper(T_DOUBLE, Matcher::vector_length(this));
+    __ vid_v(as_VectorRegister($v0$$reg));
+    __ vmv_v_x(as_VectorRegister($tmp1$$reg), $idx$$Register);
+    __ vmseq_vv(as_VectorRegister($v0$$reg), as_VectorRegister($v0$$reg), as_VectorRegister($tmp1$$reg));
+    __ vfmerge_vfm(as_VectorRegister($dst$$reg), as_VectorRegister($src$$reg), $val$$FloatRegister);
+  %}
+  ins_pipe(pipe_slow);
+%}
+
+// ------------------------------ Vector mask reductions -----------------------
+
+// true count
+
+instruct vmask_truecount(iRegINoSp dst, vRegMask src) %{
+  match(Set dst (VectorMaskTrueCount src));
+  format %{ "vmask_truecount $dst, $src" %}
+  ins_encode %{
+    BasicType bt = Matcher::vector_element_basic_type(this, $src);
+    __ vsetvli_helper(bt, Matcher::vector_length(this, $src));
+    __ vcpop_m($dst$$Register, as_VectorRegister($src$$reg));
+  %}
+  ins_pipe(pipe_slow);
+%}
+
+// first true
+
+// Return the index of the first mask lane that is set, or vector length if none of
+// them are set.
+
+instruct vmask_firsttrue(iRegINoSp dst, vRegMask src, vRegMask tmp) %{
+  match(Set dst (VectorMaskFirstTrue src));
+  effect(TEMP tmp);
+  format %{ "vmask_firsttrue $dst, $src\t# KILL $tmp" %}
+  ins_encode %{
+    BasicType bt = Matcher::vector_element_basic_type(this, $src);
+    __ vsetvli_helper(bt, Matcher::vector_length(this, $src));
+    __ vmsbf_m(as_VectorRegister($tmp$$reg), as_VectorRegister($src$$reg));
+    __ vcpop_m($dst$$Register, as_VectorRegister($tmp$$reg));
+  %}
+  ins_pipe(pipe_slow);
+%}
+
+instruct vmask_firsttrue_masked(iRegINoSp dst, vRegMask src, vRegMask_V0 v0, vRegMask tmp) %{
+  match(Set dst (VectorMaskFirstTrue src v0));
+  effect(TEMP tmp);
+  format %{ "vmask_firsttrue_masked $dst, $src, $v0\t# KILL $tmp" %}
+  ins_encode %{
+    BasicType bt = Matcher::vector_element_basic_type(this, $src);
+    __ vsetvli_helper(bt, Matcher::vector_length(this, $src));
+    __ vmsbf_m(as_VectorRegister($tmp$$reg), as_VectorRegister($src$$reg), Assembler::v0_t);
+    __ vcpop_m($dst$$Register, as_VectorRegister($tmp$$reg));
```

Comment:
Shouldn't this be: `__ vcpop_m($dst$$Register, as_VectorRegister($tmp$$reg), Assembler::v0_t);`?
And do we missed `VectorMaskLastTrue`?

Reply:
> Shouldn't this be: `__ vcpop_m($dst$$Register, as_VectorRegister($tmp$$reg), Assembler::v0_t);`? And do we missed `VectorMaskLastTrue`?

Based on our research, we have found that RISC-V does not require the addition of the `vmask_firsttrue_masked` node (currently only present in ARM64). This is because current we already have the application vector length or AVL setting in RISC-V. 
Therefore, the `VectorMaskFirstTrue` node on RISC-V does not need a partial operation to substitute vector length for vector register size, as in ARM64:
https://github.com/openjdk/jdk/blob/3c9ec26370dfae5d1230b6b69ae26122fe42e51d/src/hotspot/cpu/aarch64/aarch64_vector.ad#L290-L294
We will remove `vmask_firsttrue_masked` next.